sdk/sdk-api/src/main/java/mil/emp3/api/Square.java

```diff
@@ -91,15 +87,24 @@ public Square(IGeoSquare oRenderable) {
      * @param fValue The width in meters.
      */
     @Override
-    public void setWidth(double fValue) {
-        fValue = makePositive(fValue, "Invalid width. NaN");
-        if(this.getWidth() < MINIMUM_WIDTH) {
-            throw new IllegalArgumentException("Invalid width. " + fValue + " Minimum supported " + MINIMUM_WIDTH);
-        }
+    public void setWidth(final double fValue) {
+        validateWidth(fValue);
         this.getRenderable().setWidth(fValue);
     }
 
     /**
+     * Validates whether or not a given input is a valid Width.
+     * Throws an exception if invalid in order to inform user what the issue was.
+     * @param dValue The width to be checked
+     */
+    private void validateWidth(final double dValue) {
```

Comment:
isn't this a duplicate to another validate width in rectangle? Can you make these reference the same method.  Make sure it is protected so it isn't exposed as a method in the api

Reply:
The square and rectangle class aren't directly related except through the shared parent of Feature.  Should I bump validateWidth(test, CONSTANT) into that and then just pass in what the bounds should be? If that's the case should I generalize it a bit more to validateParameter and use that for all classes?